DZ 2/DZ_2.js

```diff
@@ -0,0 +1,145 @@
+// Первое задание. Почему код дает именно такие результаты
+var a = 1, b = 1, c, d;
+c = ++a; alert(c);           // 2 Потому что префиксная форма оператора
+d = b++; alert(d);           // 1 Потому что постфиксная форма оператора
+c = (2+ ++a); alert(c);      // 5 Сложение инкрементированного префиксного "a" который был равен 2 ,стал 3 + 2-кой   
+d = (2+ b++); alert(d);      // 4 Сложение инкрементированного постфиксного "b" который был равен 1 ,стал 2 + 2-кой
+alert(a);                    // 3 После 2 - ного инкрементирования равняется 3
+alert(b);                    // 3 После 2 - ного инкрементирования равняется 3
+
+// Второе задание 
+
+var a = 2;
+var x = 1 + (a *= 2);
+alert(x) // x будет равен 5 так как действие выполняющееся в скобках
+// имеет приоритет , а оператор *= присваивание с умножением дает 4 в данном случае
+
+// задание 3 
+//Объявить две целочисленные переменные — a и b и задать им произвольные начальные значения. Затем написать скрипт, который работает по следующему принципу:
+var x = parseInt(prompt('Введите число')) , y = parseInt(prompt('Введите 2-е число')); 
+if (x > 0 && y > 0){
```

Comment:
не хватает пробелов, проверьте здесь и далее
```suggestion
if (x > 0 && y > 0) {
```